src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/caseload/ComCaseloadSearchService.kt

```diff
@@ -69,16 +71,11 @@ class ComCaseloadSearchService(
       )
     }
 
-    val prisonersWithoutLicences = cvlSearchRecords.filter { searchRecord -> searchRecord.licence == null }.mapNotNull {
-      it.prisonerSearchPrisoner
-    }
-    val licenceStartDates = releaseDateService.getLicenceStartDates(prisonersWithoutLicences)
-
     val searchResults = cvlSearchRecords.mapNotNull {
-      val licenceStartDate = licenceStartDates[it.prisonerSearchPrisoner?.prisonerNumber]
+      val cvlRecord = cvlRecords.find { cvlRecord -> it.prisonerSearchPrisoner?.prisonerNumber == cvlRecord.nomisId }
```

Comment:
`CvlRecord` is being handled as nullable here, as the calling map is based on delius record, not prisoner search record. Therefore it's feasible that NOMIS ID could be null, and we wouldn't want to blow up here for that, but rather filter them out.

Reply:
I'm not generally a bit fan of specifying the type if its obvious based on context but this might be a good point to explicitly state it?